hack/deployer/runner/aks.go

```diff
@@ -181,9 +187,48 @@ func (d *AKSDriver) GetCredentials() error {
 
 func (d *AKSDriver) delete() error {
 	log.Print("Deleting cluster...")
-	return azure.Cmd("aks",
+	args := []string{"aks",
 		"delete", "--yes",
 		"--name", d.plan.ClusterName,
-		"--resource-group", d.plan.Aks.ResourceGroup).
-		Run()
+		"--resource-group", d.plan.Aks.ResourceGroup}
+	if !d.withoutDocker {
+		return azure.Cmd(args...).
+			Run()
+	}
+	_, err := exec.NewCommand(fmt.Sprintf("az %s", strings.Join(args, ""))).Output()
+	return err
+}
+
+func (d *AKSDriver) Cleanup() ([]string, error) {
+	// Do not use docker in the cleanup steps
+	d.withoutDocker = true
+	if err := d.auth(); err != nil {
+		return nil, err
+	}
+	daysAgo := time.Now().Add(-24 * 3 * time.Hour)
+	d.ctx["Date"] = daysAgo.Format(time.RFC3339)
+	clustersCmd := `az resource list -l {{.Location}} -g {{.ResourceGroup}} --resource-type "Microsoft.ContainerService/managedClusters" --query "[?tags.project == 'eck-ci']" | jq -r --arg d "{{.Date}}" 'map(select(.createdTime | . <= $d))|.[].name'`
```

Comment:
Shouldn't we use the same strategy as on gke and eks and filter ECK clusters by name? Imagine if we move the `e2e-monitor` cluster to aks, wouldn't it be deleted if the filter is on the tags.project?

Reply:
Good point. I've added this check to the jq piece of this command.